sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java

```diff
@@ -32,21 +30,22 @@
 
   @Override
   public SpanId generateSpanId() {
-    long id;
-    do {
-      id = random.nextLong();
-    } while (id == INVALID_ID);
+    long id = generateValidId();
     return new SpanId(id);
   }
 
   @Override
   public TraceId generateTraceId() {
-    long idHi;
-    long idLo;
-    do {
-      idHi = random.nextLong();
-      idLo = random.nextLong();
-    } while (idHi == INVALID_ID && idLo == INVALID_ID);
+    long idHi = generateValidId();
```

Comment:
This is not equivalent with the previous logic I think. The previous logic was that for trace ID the whole id should not be 0, individual parts can be. Am I missing something?

Reply:
Correct, this is a small change in logic.  Is this change problematic, or just surprising, or both?

Reply:
I'd rather keep both as zero for sanity purposes. There was been discussion in other places about what to do for vendors that to not support a 64 bit ids, and whether they should use the most significant or least significant bits, so this *could* impact them.

Reply:
Just to be clear, you mean that you'd prefer that either one may be zero, like the former logic?